ts/store/middlewares/navigation.ts

```diff
@@ -1,32 +1,23 @@
-import { NavigationState } from "react-navigation";
 import { setInstabugUserAttribute } from "../../boot/configureInstabug";
 import { mixpanelTrack } from "../../mixpanel";
 import { noAnalyticsRoutes } from "../../utils/analytics";
-import { getCurrentRouteName } from "../../utils/navigation";
 
-export const trackScreen = (
-  previousState: NavigationState,
-  currentState: NavigationState
+export const trackScreen = async (
+  previousScreen: string | undefined,
+  currentScreen: string
 ) => {
-  const previousScreenName = getCurrentRouteName(previousState);
-  const screenName = getCurrentRouteName(currentState);
-
-  if (
-    previousScreenName !== undefined &&
-    screenName !== undefined &&
-    previousScreenName !== screenName
-  ) {
-    setInstabugUserAttribute("activeScreen", screenName);
+  if (previousScreen !== currentScreen) {
+    setInstabugUserAttribute("activeScreen", currentScreen);
 
     // track only those events that are not included in the blacklist
-    if (!noAnalyticsRoutes.has(screenName)) {
-      void mixpanelTrack("SCREEN_CHANGE_V2", {
-        SCREEN_NAME: screenName
+    if (!noAnalyticsRoutes.has(currentScreen)) {
+      await mixpanelTrack("SCREEN_CHANGE_V2", {
```

Comment:
Is there a reason about the `await`? Can't we just execute this function non-blocking? Same for the other one.

Reply:
I agree, I don't think there is any added value having the async/await for the `trackScreen` method. 
Removed in 8abcc2a21f0287d9e11db9852857f9f3db5f42b9!